### ISSUE TITLE
Card ADA review

### ISSUE THREAD

**Issue by libertymayc** [jpmorganchase/salt-ds#1321]

**Comment by parsakhan21** [comment]
April 13 - ivan to address comments today

**Comment by ivan-calderon** [comment]
@maj-stella could you please advise further or be more be more specific on your recommendation here? Currently, we are treating the card as a block link, as per the [W3C description](https://design-system.w3.org/components/cards.html) but rather than 

Is your recommendation to remove the focus state of the card component itself? Then recommend that focus should be attributed to the main link, button or action within the card? TY!

**Comment by maj-stella** [comment]
Hi Ivan, thanks for picking this up and welcome to the team. Looking forward working with you!:)

Yes my recommendation was based on assumption that we want to design a card similar to cards which are used in Applications and Websites. Please see some examples what to consider about card design:

Carbon’s consideration about the different content in a tile: https://carbondesignsystem.com/components/tile/usage 

Chakra’s card: https://chakra-ui.com/docs/components/card/usage 

Here is a good article about designing/developing a card: https://css-tricks.com/block-links-the-search-for-a-perfect-solution/ 

In our design spec we have little containers without content except wording ‘Content area’. Please screenshot below. Please let me know if this is a final design for a card, a small container with short text (a word) as seen in Storybook? If so, it needs to be keyboard accessible and have a role. <div>  does not convey any semantics to a screen reader users therefore they are not aware that an element is interactive.

Thanks.


![Image](https://user-images.githubusercontent.com/112501956/232598587-eeafb7e2-597e-4fea-a24f-2cda15a281b3.png)


![Image](https://user-images.githubusercontent.com/112501956/232599906-0e96e102-0f6b-48d2-87fe-6cb79c0bbe0c.png)

**Comment by maj-stella** [comment]
Hi @ivan-calderon ,
Thank you for creating the spec which shows two different cards (Default and Interactable). ’Default’ card has interactive elements such as link and button; therefore focus follows tabbing order through the page. Interactable card does not include any interactive elements, the whole card is clickable/focusable.

I have reviewed the following example of ‘Default’ card design in StoryBook: 
https://efe4b541.saltdesignsystem-storybook.pages.dev/?path=/story/core-card--with-image-and-button

The following need to be considered/addressed:
CC @libertymayc 

- HTML elements need to have correct semantics. Semantic elements in a card on a website is important for screen reader to navigate and understand the site.
- Structure of html elements need to be considered such as where to include image? 
- Is image decorative or informative (does it need an alt text or not)?
- Ambiguous links and buttons such as ‘SEE MORE’. Usually there are more cards in a website so there would be many links with the same label. E.g. If a screen reader opens list of form elements, he/she is not able to distinguish among them as they are all labeled the same ‘SEE MORE’.

![Image](https://user-images.githubusercontent.com/112501956/232850549-32204ca2-2e27-4cba-bfd5-29066093763f.png)

**The 'Default' card can have enhanced feature so that all the card is clickable which is easier to click with a mouse and touch. Therefore there is no need to have two different cards (Default and Interactable)**

I have reviewed Interactable card design in Figma:
The focus indicator (border around the card) is too big. Users with magnification softer might not be able to perceive it. Please consider the provided IBM example in Figma. Their example shows that background colour changes. Also notice how they indicate an interactive card with an arrow.

**Comment by bhoppers2008** [comment]
@maj-stella bullets 2, 3 and 4 are only applicable to the example. They are not part of the component. Can you provide recommendations for these bullets, and the team can build them in to the example?

The component itself is simply the container. So in regard Default vs Interactable, both variants of the card can contain any elements (interactive or static). As discussed in the recent call, can you provide recommendations that should be included in the site docs for consumers, so they understand what should or should be included (for example, if you're recommending that an interactable card should ideally contain an interactive element that receives focus, and represents the action that will occur when the card is clicked). @libertymayc and @ivan-calderon can then ensure these are published on the Salt site.

Regarding the actual build... enhancement vs separate component... have you discussed the reason for the current implementation with @joshwooding and @libertymayc?

Also, regarding the IBM example, you are referring to an example of use, not the actual component Carbon provide. You'll see the have taken the same approach as us [here](https://carbondesignsystem.com/components/tile/code). The tile can receive focus (displayed with a focus ring), doesn't contain content by default, and doesn't change colour on hover. IBM regard themselves as an accessible system, do you also have concerns with their approach to the basic component?

**Comment by libertymayc** [comment]
Please see https://github.com/jpmorganchase/salt-ds/pull/1553 on improvements to the story re bullet points made above

Interactable Card is an 'enhanced' version of Card with extended props, is there a problem to have them as separate components? I would assume having it all in the same component would actually be more ambiguous e.g. setting a Default Card as disabled when it is not interactable?

**Comment by joshwooding** [comment]
@maj-stella to review comments

**Comment by parsakhan21** [comment]
April 20 - stella to revisit this ticket today; requires a few changes

**Comment by maj-stella** [comment]
@bhoppers2008 
Yes, I can confirm that bullets 2,3,4 are regarding the example. Last year it was agreed that all our examples must be accessible; therefore I report everything.

I can explain bullet point two about why the position of the image should be considered. Usually each card have a heading therefore elements which come after the heading are related to that heading (that card). If a screen reader user uses the down arrow key to read the page, he/she would know which image is related to each heading (card). But if you think  that image is only a decorative image then it doesn’t really matter where it is placed as it is not going to be announced to screen reader users.

Regarding 3 and 4 point as you are already aware this are basic considerations to take into account. 

Regarding IBM tile example I checked their code, they included everything in <a> element because the tile content is really simple, no interaction in a tile; therefore won’t cover bigger area, however I suppose our card component does. So please ignore that specific example. Check examples that Ivan has provided in the Figma file. 

Regarding your comment about IBM system and if I trust it or not, I’m just trying to explain the difference between what we have done versus what they have done. Moving forward, may I please ask that we keep our comments to each other professional. We are all in the same team and trying to get the work done in most effective manner. I can sense that there is frustration on your part but know that Im providing the most comprehensive review so nothing is missed.

I’d like to be clear here. My first suggestion was to reconsider the design so developers are clear what to build because the Figma file hasn’t provided enough clarity.

There is much more interactions to consider because card is not a shell but card, containing a content and therefore interactions need to be considered so a developer can build a card.

Ivan has done a good job with defining better clarity on interactions. As per Ivan design I suggested that a Default card (which has focus on interactive elements) could also have all clickable area (the whole card is clickable). The focus stays on an interactive element.

**Comment by bhoppers2008** [comment]
@maj-stella this is the kind of information that would be good to provide consumers in the site documentation. If there is more of this guidance you'd like to provide consumers can you ensure liberty and ivan are aware.
Regarding IBM. I'm afraid i don't understand your comment. We're effectively delivering the same component. Ivan's examples are illustrating consumer use in the same way IBM's does. Out component should be considered in the same way as IBMs.
Regarding comments to each other. This isn't the right forum. I'll setup time to discuss.

**Comment by ivan-calderon** [comment]
Thank you for your time today @maj-stella! It was a good chat. I just wanted to recap what we discussed on our call today:

1. We should provide accessibility recommendations to our consumers about how to use both variants, default and interactable cards. 
2. We are treating card as a base component. However, we will add examples of how to use and populate card to make it accessible.  
3. Default card can host interactive elements. Where interactions, including focus, should point to those elements for accessibility.   
4. Interactable card **shouldn't** host interactive elements into the card, and focus state should point to the card as a block link.

Actions:

- Stella: Share accessibility recommendations for both, Default and Interactable card variants.
- Liberty: Add examples on Github  
- Ivan: Implement changes, if any, reflecting accessibility recommendations.

cc. @libertymayc, @shey-v, @bhoppers2008, @joshwooding   

Please feel free to add or correct actions here. TY team!

**Comment by libertymayc** [comment]
Examples added in this PR: https://github.com/jpmorganchase/salt-ds/pull/1553
